src/app/common/services/project-service.coffee

```diff
@@ -29,10 +29,16 @@ angular.module("doubtfire.common.services.projects", [])
     ]
 
     for t in times
-      diff = laterTime.diff(earlyTime, t)
-      if diff > 1
+      #exactDiff is floating point
+      exactDiff = laterTime.diff(earlyTime, t, true).toFixed(2)
```

Comment:
Why are we diffing as a floating point, and then immediately flooring to an integer? Can you explain what's going on? Cheers.

Reply:
So, the diff in already existing code returns a truncated number i.e. zero decimal places. Adding a flag true returns a floating point number but when I tried this, various plagiarism issues starting occurring out of nowhere (although this function is not used anywhere else). So, I didn't want to break the old code I added a new field exactDiff that is floating point. In case of old code, if the number of days is 20 then it would return diff as 2 weeks due to truncated value and then it will show 20 days (as we are checking diff > 2) and not 2 weeks. To correct this error I need the floating value to check if it's greater than 14 days then show as weeks.